Runtime/World/World.h

```diff
@@ -91,4 +114,28 @@ namespace Spartan
 		Scene_State m_state;
 		std::string m_name;
 	};
+
+    template<typename T>
+    inline std::shared_ptr<ComponentManager<T>> World::GetComponentManager()
+    {
+        VALIDATE_COMPONENT_TYPE(T);
+        ComponentType type = IComponent::TypeToEnum<T>();
+
+        std::shared_ptr<BaseComponentManager> manager = m_components_managers[type];
+
+        return std::static_pointer_cast<ComponentManager<T>>(manager);
+    }
+
+    template<typename Func>
+    inline void World::IterateManagers(Func f)
+    {
+        // Hardcoded types make it harder to write new components
+        Spartan::for_each(std::tie(Transform(), AudioSource(), AudioListener(),
```

Comment:
Hardcoded types mean that when a coder writes a new component, he'll have to visit all of these places and update them. This makes maintenance more difficult.
Do you believe there is something we can do about it ?

Reply:
Again the only thing that comes to mind is creating a global tuple that would be shared by this function the one I will make to automate the proccess of creating the managers.

Reply:
[This](https://github.com/PanosK92/SpartanEngine/blob/77fd20693244a47735f391f6efa95015a155727f/Runtime/World/Components/IComponent.cpp#L70) is where I associate the component type with the enumeration. You can add your global tuple there if you wish. This way we will know that whenever we want to write a new component, we only have to visit one place, IComponent.